src/body/body.rs

```diff
@@ -628,7 +651,7 @@ mod tests {
 
         assert_eq!(
             mem::size_of::<Sender>(),
-            mem::size_of::<usize>() * 4,
+            mem::size_of::<usize>() * 5,
```

Comment:
This was the main thing I was a little worried about when filing the issue... Can you tell in benchmarks how much the increased size (and perhaps the creation of a `oneshot` for each request) effects things? If it's nothing, cool. But it'd be good to know for sure.